### ISSUE TITLE
Hypersonic doing too much damage

### ISSUE THREAD

**Issue by LemADEC** [BuiltBrokenModding/ICBM-Classic#52]
As of ICBM-classic-1.7.10-2.15.1b153 / VoltzEngine-1.7.10-1.3.0b82, hypersonic explosive can break through 10+ layers of obsidian.
By comparison, endo/exothermic explosive give a much lower radius of 5 blocks inside obsidian (still a bit too strong in my opinion).
By extension, it can also break through WarpDrive forcefield and hull blocks like butter.
For the record, exothermic explosive seems to ignore block hardness and just do radius effect, which isn't good for bunker blocks.

For the record, both IC2, GregTech and WarpDrive are adjusting block blast resistances:
- Obsidian defaults to 6000 blast resistance, adjusted down to 180
- Anvil & Enchanting table defaults to 6000, adjusted down to 5
- Ender chest defaults to 3000, adjusted down to 180
- Fluid defaults to 500, adjusted down to 90

**Comment by LemADEC** [comment]
Hypersonic is also using a radius effect instead of pathfinding, which isn't good for bunker blocks (same as Exothermic). That issue might be generic.

**Comment by DarkGuardsman** [comment]
Ya none of the old ICBM explosives are good for bunkers. They all use iterative loops with no pathing at all. The closest thing they have is some stop blocks when they hit hard surfaces but that doesn't perfectly block explosives.

**Comment by LemADEC** [comment]
As of ICBM-1.2.1.172 (Voltz with patched GC), Nuclear was doing pathfinding and bunker support (that's 2 layers of Reinforced concrete):
![image](https://cloud.githubusercontent.com/assets/4776032/26572835/b1873854-451b-11e7-9d18-42734fc3115a.png)
It had memory allocation issues causing major lag, but that's another story.

On the same version, Hypersonic shows a form or pathfinding effect depending on materials:
![image](https://cloud.githubusercontent.com/assets/4776032/26573390/e5117f02-451d-11e7-8247-bbad164aef5d.png)
Effect goes up to 13 blocks away of stone.
It doesn't affect obsidian:
![image](https://cloud.githubusercontent.com/assets/4776032/26574197/ba170df0-4520-11e7-9198-e28ad71e31e8.png)


As of ICBM-classic-1.7.10-2.15.1b162 & VoltzEngine-1.7.10-1.4.0b95 (with ICBM and WorldEdit, no other mods):
before:
![image](https://cloud.githubusercontent.com/assets/4776032/26573804/5e2280d4-451f-11e7-9269-014f0a561468.png)
after:
![image](https://cloud.githubusercontent.com/assets/4776032/26573833/75b44138-451f-11e7-95a1-ee7c2f97733a.png)
Effect goes up to ~8 blocks away of stone.
It also eats up 40 blocks deep of obsidian in a square shape.

**Comment by DarkGuardsman** [comment]
1.5.2 still didn't have any pathfinding as I worked on the code. Its uses the same iterative loop system in both 1.5 and 1.6. That being said I do plan to rewrite it to use the new ICBM's path code. So it shouldn't be a problem.

**Comment by DarkGuardsman** [comment]
The issue is solved, looks like some patch code was added at one point to destroy blocks of hardness above the energy of blast. Unsure if this is by design or a bug... but I removed the code to solve the issues. However, this has created an issue with flying blocks hitting the obsidian and getting stuck. 

Additionally, I reverse engineers the system used for the blasts. Its a rotation based ray trace system. It will go 2 degrees at a time in a ball shape then out to the distance of the radius. This results in a star pattern giving the illusion of pathfinding. I will be uploading a video later that shows me reverse engineering the code and explaining how it works.
